# **연휴 대체과제**
- 추석 연휴 대체과제는 **파이썬 연습과제**입니다.
- 코드 작성하시고, 출력 결과까지 나오도록 실행 부탁드립니다.
  - 제출 시 파일명 본인 이름으로 변경해 주세요. ex) Week 0_대체과제_OOO
  - 해당 파일에 문제 푸신 후 제출해 주시면 됩니다.
- 과제는 팀원들과 같이 풀이하셔도 좋고, 책을 참고해도 좋고, ChatGPT를 써도 좋고, 구글링을 해도 되고 다 좋습니다.
  - 정규 세션 전 가볍게 파이썬 기본 문법 연습하는 느낌으로 공부해주세요 :)
  - 백지만 내지 말아주세요..😥

## **1. 파이썬 자료구조**

다음과 같이 출력 되도록 코드를 작성하세요

### Q1: Type 확인

In [ ]:
a = 3.14

타입을 확인해 주세요

In [ ]:
# 타입 확인
print(type(a).__name__)

float


<p><strong>[출력 결과]</strong></p><pre>float</pre>

### Q2: Type 변경

In [ ]:
a = 1

타입을 변경해 주세요

In [ ]:
# 타입 변경
a = float(a)
print(a)

1.0


<p><strong>[출력 결과]</strong></p><pre>1.0</pre>

### Q3: Type 변경

In [ ]:
a = 1

타입을 변경해 주세요

In [ ]:
# 타입 변경
a = bool(a)
print(a)

True


<p><strong>[출력 결과]</strong></p><pre>True</pre>

### Q4: 타입 변경

In [ ]:
a = False

타입을 변경해 주세요

In [ ]:
# 타입 변경
a = int(a)
print(a)

0


<p><strong>[출력 결과]</strong></p><pre>0</pre>

### Q5: 타입 변경

In [ ]:
a = 123

타입을 변경해 주세요

In [ ]:
# 타입변경
a = str(a)
print(repr(a))

'123'


<p><strong>[출력 결과]</strong></p><pre>'123'</pre>

## **2. ⭐ 리스트, 튜플**

### 연습문제 1

In [ ]:
movie = ['어벤져스', '아이언맨', '토르', '스파이더맨']

`movie`에 맨 끝에 `엑스맨` 을 추가하세요.

In [ ]:
movie.append('엑스맨')
print(movie)

['어벤져스', '아이언맨', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '아이언맨', '토르', '스파이더맨', '엑스맨']</pre>

`movie`의 1번 index에 `데드풀`을 추가하세요

In [ ]:
movie.insert(1,'데드풀')
print(movie)

['어벤져스', '데드풀', '아이언맨', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '데드풀', '아이언맨', '토르', '스파이더맨', '엑스맨']</pre>

`movie`에서 `아이언맨`을 삭제 하세요.

- remove() 사용

In [ ]:
movie.remove('아이언맨')
print(movie)

['어벤져스', '데드풀', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '데드풀', '토르', '스파이더맨', '엑스맨']</pre>

`movie`에서 1번째 index를 삭제하세요

In [ ]:
del movie[1]
print(movie)

['어벤져스', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '토르', '스파이더맨', '엑스맨']</pre>

In [ ]:
kr_movie = ['승리호', '신세계', '타짜']

`kr_movie`를 `movie`에 추가해 주세요

In [ ]:
movie.extend(kr_movie)
print(movie)


['어벤져스', '토르', '스파이더맨', '엑스맨', '승리호', '신세계', '타짜']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '토르', '스파이더맨', '엑스맨', '승리호', '신세계', '타짜']</pre>

`movie`를 가나다 순으로 정렬하세요

In [ ]:
movie.sort()
print(movie)

['스파이더맨', '승리호', '신세계', '어벤져스', '엑스맨', '타짜', '토르']


<p><strong>[출력 결과]</strong></p><pre>['스파이더맨', '승리호', '신세계', '어벤져스', '엑스맨', '타짜', '토르']</pre>

### 연습문제 2

In [ ]:
fruit = ['사과', '바나나', '파인애플', '배', '수박', '키위', '오렌지', '망고', '딸기']

list **인덱싱** & **슬라이싱**을 사용하여 다음과 같이 출력 하세요

In [ ]:
result = fruit[2:5]
print(result)

['파인애플', '배', '수박']


<p><strong>[출력 결과]</strong></p><pre>['파인애플', '배', '수박']</pre>

In [ ]:
result = fruit[7:9]
print(result)

['망고', '딸기']


<p><strong>[출력 결과]</strong></p><pre>['망고', '딸기']</pre>

In [ ]:
result = fruit[0:3]
print(result)


['사과', '바나나', '파인애플']


<p><strong>[출력 결과]</strong></p><pre>['사과', '바나나', '파인애플']</pre>

In [ ]:
result = fruit[3:9]
print(result)

['배', '수박', '키위', '오렌지', '망고', '딸기']


<p><strong>[출력 결과]</strong></p><pre>['배', '수박', '키위', '오렌지', '망고', '딸기']</pre>

### 연습문제 3

슬라이싱을 활용하여 **짝수**만 출력해 주세요

In [ ]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [ ]:
even_nums = nums[1::2]
print(even_nums)

[2, 4, 6, 8, 10]


<p><strong>[출력 결과]</strong></p><pre>[2, 4, 6, 8, 10]
</pre>

### 연습문제 4

sample = [1, 2, 3, 4, 5, 6, 7, 8] 리스트를 슬라이싱을 이용하여 출력과 같이 나올수 있도록 코드를 채워주세요.

In [ ]:
sample = [1, 2, 3, 4, 5, 6, 7, 8]

슬라이싱을 사용해 다음과 같이 출력해 주세요

In [ ]:
reversed_sample = sample[::-1]
print(reversed_sample)

[8, 7, 6, 5, 4, 3, 2, 1]


<p><strong>[출력 결과]</strong></p><pre>[8, 7, 6, 5, 4, 3, 2, 1]</pre>

슬라이싱을 사용해 다음과 같이 출력해 주세요

In [ ]:
sliced_sample = sample[2::2]
print(sliced_sample)

[3, 5, 7]


<p><strong>[출력 결과]</strong></p><pre>[3, 5, 7]</pre>

슬라이싱을 사용해 다음과 같이 출력해 주세요

In [ ]:
sliced_sample = sample[1::3]
print(sliced_sample)

[2, 5, 8]


<p><strong>[출력 결과]</strong></p><pre>[2, 5, 8]</pre>

### 연습문제 5

- 아래의 값을 저장하고 있는 `myList` 변수를 생성하세요.
- [[1,2,3],[4,5,6],[7,8]]

In [ ]:
myList = [[1,2,3],[4,5,6],[7,8]]
myList

[[1, 2, 3], [4, 5, 6], [7, 8]]

- 9를 `myList`에 추가하여 아래의 값을 저장할 수 있도록 하세요.
    - [[1,2,3],[4,5,6],[7,8,9]]

In [ ]:
myList[-1].append(9)
myList

[[1, 2, 3], [4, 5, 6], [7, 8, 9]]

<p><strong>[출력 결과]</strong></p><pre>[[1, 2, 3], [4, 5, 6], [7, 8, 9]]</pre>

### 연습문제 6

주어진 tuple `a`를 (1, 100, 2, 3, 4)의 요소를 가지도록 변경해 주세요

In [ ]:
a = (1, 2, 3, 4)

In [ ]:
a = (1,100) + a[1:]
print(a)

(1, 100, 2, 3, 4)


<p><strong>[출력 결과]</strong></p><pre>(1, 100, 2, 3, 4)</pre>

## **3. 셋, 딕셔너리**

### 연습문제 1

In [ ]:
sample1 = {1, 2, 3, 4, 5}
sample2 = {2, 4, 5, 6, 7}

print(f'sample1: {sample1}')
print(f'sample2: {sample2}')

sample1: {1, 2, 3, 4, 5}
sample2: {2, 4, 5, 6, 7}


`sample1` 에 6를 추가합니다.

In [ ]:
sample1.add(6)
print(sample1)

{1, 2, 3, 4, 5, 6}


<p><strong>[출력 결과]</strong></p><pre>{1, 2, 3, 4, 5, 6}</pre>

`sample2`에 2를 제거합니다.

In [ ]:
sample2.discard(2)
print(sample2)

{4, 5, 6, 7}


<p><strong>[출력 결과]</strong></p><pre>{4, 5, 6, 7}</pre>

`sample1`과 `sample2`의 **교집합**을 출력합니다.

In [ ]:
intersection = sample1 & sample2
print(intersection)

{4, 5, 6}


<p><strong>[출력 결과]</strong></p><pre>{4, 5, 6}</pre>

`sample1`과 `sample2`의 **합집합**을 출력합니다.

In [ ]:
union = sample1 | sample2
print(union)

{1, 2, 3, 4, 5, 6, 7}


<p><strong>[출력 결과]</strong></p><pre>{1, 2, 3, 4, 5, 6, 7}</pre>

`sample1`과 `sample2`의 **차집합**을 활용하여 다음과 같이 출력합니다.

In [ ]:
difference = sample1 - sample2
print(difference)

{1, 2, 3}


<p><strong>[출력 결과]</strong></p><pre>{1, 2, 3}</pre>

### 연습문제 2

- 다음 리스트에서 중복된 항목을 제거하세요
- 최종 출력 값은 list 형태로 출력하세요

In [ ]:
numList = [1, 3, 2, 3, 7, 6, 8, 4, 10, 5, 3, 8, 9]

In [ ]:
newList = list(set(numList))
newList

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

<p><strong>[출력 결과]</strong></p><pre>[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]</pre>

### 연습문제 3

- 학생들의 이름과 점수를 key와 value로 가지는 `score` 변수를 생성하세요.
    - 하준 90점, 서윤 86점, 지아 80점

In [ ]:
score = {
    '하준': 90,
    '서윤': 86,
    '지아': 80
}
print(score)

{'하준': 90, '서윤': 86, '지아': 80}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '지아': 80}</pre>

score 변수에 수지 95점을 추가하세요.

In [ ]:
score['수지'] = 95
print(score)

{'하준': 90, '서윤': 86, '지아': 80, '수지': 95}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '지아': 80, '수지': 95}</pre>

score 변수에서 지아를 삭제하세요.

In [ ]:
del score['지아']
print(score)

{'하준': 90, '서윤': 86, '수지': 95}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '수지': 95}</pre>

score 변수에 기창 98점, 남철 60점, 기성 75점을 **한번에** 추가하세요.
- update() 를 사용하세요

In [ ]:
new_score = {
    '기창': 98,
    '남철': 60,
    '기성': 75
}
score.update(new_score)
print(score)

{'하준': 90, '서윤': 86, '수지': 95, '기창': 98, '남철': 60, '기성': 75}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '수지': 95, '기창': 98, '남철': 60, '기성': 75}</pre>

## **4. 문자열**

### 연습문제 1

- 변수 `a`를 활용하여 문자열 포맷팅으로 출력하세요
- print() 를 사용합니다.

In [ ]:
a = 1234

In [ ]:
print(f'a의 값은 {a}입니다')


a의 값은 1234입니다


<p><strong>[출력 결과]</strong></p><pre>a의 값은 1234입니다
</pre>

- 변수 `a`, `b`를 활용하여 문자열 포맷팅으로 출력하세요
- 단, `a`, `b`의 값은 소수 3째짜리까지 출력하고, 그 이하는 반올림합니다.

In [ ]:
a = 3.141592
b = 6.181112

In [ ]:
print(f'a의 값은 {a:.3f}, b의 값은 {b:.3f}')


a의 값은 3.142, b의 값은 6.181


<p><strong>[출력 결과]</strong></p><pre>a의 값은 3.142, b의 값은 6.181입니다
</pre>

### 연습문제 2

문자열 "210101-1234567"을 인덱싱 하여 성별을 나타내는 숫자를 출력 하세요

In [ ]:
registration_number =  '210101-1234567'

In [ ]:
gender = registration_number[7]
print(f"'{gender}'")

'1'


<p><strong>[출력 결과]</strong></p><pre>'1'</pre>

### 연습문제 3

문자열 '20210101Sunny'를 슬라이싱하여 날짜는 `date`변수에 날씨는 `weather` 변수에 저장 후 출력 하세요

In [ ]:
sample = '20210101Sunny'

In [ ]:
date = sample[:8]
weather = sample[8:]
print(f'date: {date}, weather: {weather}')

date: 20210101, weather: Sunny


<p><strong>[출력 결과]</strong></p><pre>date: 20210101, weather: Sunny
</pre>

(문자열 슬라이싱을 활용)자동차 번호가 다음과 같을 때 뒤에 4자리만 출력하세요.

In [ ]:
license_plate = "13나 5645"

In [ ]:
last4 = license_plate[-4:]
print(last4)

5645


<p><strong>[출력 결과]</strong></p><pre>5645
</pre>

### 연습문제 4

- 다음 중 a, b, c에서 공통으로 사용된 알파벳(공백 포함)을 출력하세요.
- list 형식으로 출력하세요.

In [ ]:
a = 'Hello, World'
b = 'World To World'
c = 'world to world'

In [ ]:
# 코드를 입력해 주세요
set_a = set(a)
set_b = set(b)
set_c = set(c)

common_char = set_a & set_c
common_char_list = sorted(common_char)

print(common_char_list)

[' ', 'd', 'l', 'o', 'r']


<p><strong>[출력 결과]</strong></p><pre>[' ', 'd', 'l', 'o', 'r']</pre>

### 연습문제 5

sample = ['Python', 'is', 'too', 'interesting']

리스트를 'Python * is * too * interesting' 이라는 문자열로 출력하세요.

In [ ]:
sample = ['Python', 'is', 'too', 'interesting']

In [ ]:
# 코드를 입력해 주세요
result = ' * '.join(sample)
print(f"'{result}'")

'Python * is * too * interesting'


<p><strong>[출력 결과]</strong></p><pre>'Python * is * too * interesting'</pre>

phone_number = ['010', '1234', '5678']

리스트를 '010-1234-5678' 이라는 문자열로 출력하세요.

In [ ]:
phone_number = ['010', '1234', '5678']

In [ ]:
# 코드를 입력해 주세요
phone_num = '-'.join(phone_number)
print(f"'{phone_num}'")

'010-1234-5678'


<p><strong>[출력 결과]</strong></p><pre>'010-1234-5678'</pre>

## **5. 연산**

### 연습문제

`a`와 `b`를 덧셈하여 다음의 결과가 나오도록 코드를 입력해 주세요

- 단, `a`와 `b`의 값을 수정하지 않습니다.

In [ ]:
a = '10'
b = 20

In [ ]:
# 코드를 입력해 주세요
result = int(a) + b
print(result)

30


<p><strong>[출력 결과]</strong></p><pre>30</pre>

In [ ]:
# 코드를 입력해 주세요
result = a + str(b)
print(f"'{result}'")

'1020'


<p><strong>[출력 결과]</strong></p><pre>'1020'</pre>

## **6. 비교/논리/삼항연산자, ⭐⭐조건문**

### 연습문제 1

사용자에게 몸무게를 입력 받아 과체중인지 정상 체중인지 출력하는 프로그램을 만드세요.

- 몸무게가 100kg 초과이면 '귀하는 과체중 입니다.' 메시지를 출력하세요
- 몸무게가 100kg 이하이면 '귀하는 정상체중 입니다.' 메시지를 출력하세요
- 몸무게 입력은 input() 함수를 사용하여 입력 받습니다.
- 반드시 **삼항연산자를 활용**하세요

In [ ]:
# 입력 예시 (105로 입력한 경우)
weight = input('몸무게를 입력해 주세요: ')

몸무게를 입력해 주세요: 105


In [ ]:
# 코드를 입력해 주세요
weight = float(weight)
message = '귀하는 과체중 입니다.' if weight > 100 else '귀하는 정상체중 입니다.'
print(f"'{message}'")

'귀하는 과체중 입니다.'


<p><strong>[출력 결과]</strong></p><pre>'귀하는 과체중 입니다.'</pre>

### 연습문제 2

#### Q1

- (몸무게 + 100 - 키)를 비만 수치 공식이라고 가정합니다.
- 키와 몸무게를 입력받아 첫 번째 줄에 비만 수치를 출력합니다.
- 비만 수치가 0보다 크면 비만이라는 메시지를 출력하세요.

```
입력 예) info =[155, 60] (키, 몸무게 순)
출력[1]: 비만수치는 5 입니다.
출력[2]: 당신은 비만입니다.
```

In [ ]:
info = [155, 60]

In [ ]:
# 코드를 입력해 주세요
height = info[0]
weight = info[1]

obesity_index = weight + 100 - height
print(f'비만수치는 {obesity_index}입니다')

if obesity_index > 0:
  print('당신은 비만입니다.')

비만수치는 5입니다
당신은 비만입니다.


<p><strong>[출력 결과]</strong></p><pre>비만수치는 5입니다
당신은 비만입니다.
</pre>

#### Q2

성적을 입력받아 90점 이상이면 A, 80점이상이면 B, 70점 이상이면 C, 70점 미만이면 D를 출력하세요
- 단, 점수가 음수이거나 100점을 초과하는 경우에는 '에러'를 출력하세요

힌트) input() 함수를 사용합니다.
- input() 함수는 사용자로부터 입력 값을 받을 수 있습니다.

In [ ]:
# 코드를 입력해 주세요
score = float(input('점수를 입력해 주세요: '))

if score < 0 or score > 100:
  print('에러')
elif score >= 90:
  print('A')
elif score >= 80:
  print('B')
elif score >= 70:
  print('C')
else:
  print('D')

점수를 입력해 주세요: 95
A


<p><strong>[출력 결과]</strong></p><pre>점수를 입력해 주세요: 95
A
</pre>

- 95점 입력시: A
- 75점 입력시: C
- -10점 입력시: 에러
- 105점 입력시: 에러

#### Q3

사용자로부터 `달러`, `엔`, `유로`, 또는 `위안` 금액을 입력받은 후 이를 `원`으로 변환하는 프로그램을 작성해 주세요.

각 통화별 환율은 다음과 같다.

- 달러: 1237.0
- 엔: 105.4
- 유로: 1349.69
- 위안: 194.57

사용자는 10 달러, 200 엔, 30 유로, 100 위안과 같이 금액과 통화명 사이에 **공백**을 넣어 입력한다고 가정합니다.

입력 예시) `10 달러`

결과 예시) `11670.0원`

In [ ]:
# 코드를 입력해 주세요
money_input = input('입력: ')

# 입력을 공백 기준으로 분리
amount_str, currency = money_input.split()

# 금액 실수로 변환
amount = float(amount_str)

# 환율
exchange_rates = {
    '달러': 1237.0,
    '엔': 105.4,
    '유로': 1349.69,
    '위안': 194.57
}

# 환율 계산
converted_amount = amount * exchange_rates[currency]

print(f'{converted_amount} 원')

입력: 10 달러
12370.0 원


<p><strong>[출력 결과]</strong></p><pre>입력: 10 달러
12370.0 원
</pre>

## **⭐⭐ 7. 반복문**

### 연습문제 1

반복문과 조건문을 활용하여 `sample` 에서 음수 값만 출력해 주세요

In [ ]:
sample = [1, 11, 22, -20, 55, -33, 0, -5, 12, 35, 19, -9, -1, -100, 90]

In [ ]:
# 코드를 입력해 주세요
for num in sample:
  if num < 0:
    print(num)

-20
-33
-5
-9
-1
-100


<p><strong>[출력 결과]</strong></p><pre>-20
-33
-5
-9
-1
-100
</pre>

반복문과 조건문을 활용하여 `sample`의 요소 중 10보다 크고 50보다 작은 요소만 출력해 주세요

In [ ]:
# 코드를 입력해 주세요
for num in sample:
  if 10 < num < 50:
    print(num)

11
22
12
35
19


<p><strong>[출력 결과]</strong></p><pre>11
22
12
35
19
</pre>

### 연습문제 2

다음의 문자열을 **반복문**을 활용하여 출력하되, **모음('a', 'e', 'i', 'o', 'u')만 대문자**로 자음은 소문자로 출력해 주세요.

In [ ]:
sample = "PyThoN Is fUn"

In [ ]:
# 코드를 입력해 주세요
#     if i.lower() in ['a', 'e', 'i', 'o', 'u']:
for i in sample:
  if i == ' ':
    print('(공백)')
  elif i.lower() in ['a', 'e', 'i', 'o', 'u']:
    print(f'(모음){i.upper()}')
  else:
    print(i.lower())

p
y
t
h
(모음)O
n
(공백)
(모음)I
s
(공백)
f
(모음)U
n


<p><strong>[출력 결과]</strong></p><pre> p
 y
 t
 h
(모음) O
 n
(공백)
(모음) I
 s
(공백)
 f
(모음) U
 n
</pre>

### 연습문제 3

- 구구단을 반복문을 사용하여 만들어 보세요!
- 구구단은 2단 부터 9단까지 있습니다.

In [ ]:
# 코드를 입력해 주세요
for i in range(2, 10):
  for j in range(1,10):
    print(f'{i} X {j} = {i * j}')
  print()

2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9
3 X 4 = 12
3 X 5 = 15
3 X 6 = 18
3 X 7 = 21
3 X 8 = 24
3 X 9 = 27

4 X 1 = 4
4 X 2 = 8
4 X 3 = 12
4 X 4 = 16
4 X 5 = 20
4 X 6 = 24
4 X 7 = 28
4 X 8 = 32
4 X 9 = 36

5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45

6 X 1 = 6
6 X 2 = 12
6 X 3 = 18
6 X 4 = 24
6 X 5 = 30
6 X 6 = 36
6 X 7 = 42
6 X 8 = 48
6 X 9 = 54

7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63

8 X 1 = 8
8 X 2 = 16
8 X 3 = 24
8 X 4 = 32
8 X 5 = 40
8 X 6 = 48
8 X 7 = 56
8 X 8 = 64
8 X 9 = 72

9 X 1 = 9
9 X 2 = 18
9 X 3 = 27
9 X 4 = 36
9 X 5 = 45
9 X 6 = 54
9 X 7 = 63
9 X 8 = 72
9 X 9 = 81



<p><strong>[출력 결과]</strong></p><pre>2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9
3 X 4 = 12
3 X 5 = 15
3 X 6 = 18
3 X 7 = 21
3 X 8 = 24
3 X 9 = 27

4 X 1 = 4
4 X 2 = 8
4 X 3 = 12
4 X 4 = 16
4 X 5 = 20
4 X 6 = 24
4 X 7 = 28
4 X 8 = 32
4 X 9 = 36

5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45

6 X 1 = 6
6 X 2 = 12
6 X 3 = 18
6 X 4 = 24
6 X 5 = 30
6 X 6 = 36
6 X 7 = 42
6 X 8 = 48
6 X 9 = 54

7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63

8 X 1 = 8
8 X 2 = 16
8 X 3 = 24
8 X 4 = 32
8 X 5 = 40
8 X 6 = 48
8 X 7 = 56
8 X 8 = 64
8 X 9 = 72

9 X 1 = 9
9 X 2 = 18
9 X 3 = 27
9 X 4 = 36
9 X 5 = 45
9 X 6 = 54
9 X 7 = 63
9 X 8 = 72
9 X 9 = 81

</pre>

### 연습문제 4

사용자의 입력이 'MALE', 'FEMALE'인 경우는 while 문을 종료하고, 그렇지 않으면 "잘못된 입력 값입니다."를 출력하세요

- input() 함수로 사용자로부터 입력을 받습니다.

In [ ]:
# 코드를 입력해 주세요
while True:
  gender_input = input()

  if gender_input in ['MALE', 'FEMALE']:
    print('====\n프로그램 종료')
    break
  else:
    print('잘못된 입력 값입니다.')


male
잘못된 입력 값입니다.
animal
잘못된 입력 값입니다.
FEMALE
====
프로그램 종료


<p><strong>[출력 결과]</strong></p><pre>male
잘못된 입력 값입니다.
animal
잘못된 입력 값입니다.
FEMALE
====
프로그램 종료
</pre>

### 연습 문제 5

1부터 50까지의 숫자 중에서 짝수의 합을 구하세요 (while 사용)

In [ ]:
# 코드를 입력해 주세요
count = 0
total = 0
num = 2

while num <= 50:
  total += num
  count += 1
  print(f'count: {count*2}, total: {total}')
  num += 2

count: 2, total: 2
count: 4, total: 6
count: 6, total: 12
count: 8, total: 20
count: 10, total: 30
count: 12, total: 42
count: 14, total: 56
count: 16, total: 72
count: 18, total: 90
count: 20, total: 110
count: 22, total: 132
count: 24, total: 156
count: 26, total: 182
count: 28, total: 210
count: 30, total: 240
count: 32, total: 272
count: 34, total: 306
count: 36, total: 342
count: 38, total: 380
count: 40, total: 420
count: 42, total: 462
count: 44, total: 506
count: 46, total: 552
count: 48, total: 600
count: 50, total: 650


<p><strong>[출력 결과]</strong></p><pre>count: 2, total: 2
count: 4, total: 6
count: 6, total: 12
count: 8, total: 20
count: 10, total: 30
count: 12, total: 42
count: 14, total: 56
count: 16, total: 72
count: 18, total: 90
count: 20, total: 110
count: 22, total: 132
count: 24, total: 156
count: 26, total: 182
count: 28, total: 210
count: 30, total: 240
count: 32, total: 272
count: 34, total: 306
count: 36, total: 342
count: 38, total: 380
count: 40, total: 420
count: 42, total: 462
count: 44, total: 506
count: 46, total: 552
count: 48, total: 600
count: 50, total: 650
</pre>

### 연습 문제 6

**while**을 이용하여 다음과 같이 @ 찍기에 도전해 보세요!

In [1]:
# 코드를 입력해 주세요
i = 1

while i <= 5:
  print('@' * i)
  i += 1

@
@@
@@@
@@@@
@@@@@


<p><strong>[출력 결과]</strong></p><pre>@
@@
@@@
@@@@
@@@@@
</pre>

In [3]:
# 코드를 입력해 주세요
i = 1

while i <= 3:
  print('@' * i)
  i += 1

i -= 2
while i > 0:
  print('@' * i)
  i -= 1

@
@@
@@@
@@
@


<p><strong>[출력 결과]</strong></p><pre>@
@@
@@@
@@
@
</pre>

## **⭐ 8. Comprehension**
- 별도의 연습문제는 없지만 중요한 파트이니 실습파일 공부하시는 걸 !적극! 추천드립니다😊

## **⭐⭐⭐ 9. 함수**

### 연습문제 1

알파벳 문자열을 인수로 넣으면  함수를 구현하세요

- 함수의 이름은 `capital_letter` 로 선언합니다.
- 함수는 **letter**라는 위치인수를 가집니다.
- 함수는 **letter**인수에 받은 문자열을 모두 대문자로 바꾸고, 바꾼 값을 `return` 합니다.

In [5]:
# 코드를 입력해 주세요
def capital_letter(letter):
  return letter.upper()

result = capital_letter('My name is Elsa')
print(f"'{result}'")

'MY NAME IS ELSA'


<p><strong>[출력 결과]</strong></p><pre>'MY NAME IS ELSA'</pre>

### 연습문제 2

def calc_many(choice, *args) 함수에서
- choice가 `sum` 이면 **합**을 구해서 return 합니다.
- choice 가 `mul` 이면 **곱**한 결과를 return 합니다.

함수를 구현하세요

In [6]:
# 코드를 입력해 주세요
def calc_many(choice,*args):
  if choice == 'sum':
    return sum(args)
  elif choice == 'mul':
    result = 1
    for num in args:
      result *= num
    return result

In [7]:
# 결과: 10
print(calc_many('sum', 0, 1, 2, 3, 4))
# 결과: 120
print(calc_many('mul', 1, 2, 3, 4, 5))

10
120


## **10. 파이썬 내장함수**

### 연습문제

`zip()`과 반복문을 활용하여 다음과 같이 출력해 주세요

In [8]:
fruit = ['사과', '자두', '딸기', '망고']
price = [1000, 3000, 2500, 6000]

In [9]:
# 코드를 입력해 주세요
for f, p in zip(fruit, price):
  print(f"({f} / {p} 원)")

(사과 / 1000 원)
(자두 / 3000 원)
(딸기 / 2500 원)
(망고 / 6000 원)


<p><strong>[출력 결과]</strong></p><pre>(사과 / 1000 원)
(자두 / 3000 원)
(딸기 / 2500 원)
(망고 / 6000 원)
</pre>